internal/features/features.go

```diff
@@ -29,15 +29,24 @@ const (
 	// the last revision is still the same at the target repository,
 	// and if that is so, skips the reconciliation.
 	OptimizedGitClones = "OptimizedGitClones"
+	// CacheSecrets controls whether secrets should be cached.
+	//
+	// Source Controller uses secrets for git auth, but an unmodified controller-runtime client will
+	// fetch and cache all secrets in the cluster. We disable that behavior by default to reduce
+	// RAM usage and RBAC requirements, but enabling this flag restores the secret-caching behavior.
+	CacheSecrets = "CacheSecrets"
```

Comment:
We have decided this should be the same across controllers for consumer ease, see the referenced issue.